share/picongpu/pypicongpu/template/include/picongpu/param/density.param.mustache

```diff
@@ -37,33 +37,82 @@ namespace picongpu
+             * generate the initial macroparticle position for species "{{{placed_species_initial.name}}}" ({{{placed_species_initial.typename}}})
+             *
+             * (Note: This profile assumes ratio to be 1. The correct ratio is applied in the initpipeline by CreateDensity.)
+             */
+            struct init_{{{placed_species_initial.typename}}}_functor
+            {
+                HDINLINE float_X operator()(const floatD_64& position_SI, const float3_64& cellSize_SI)
                 {
-                    HDINLINE float_X operator()(const floatD_64& position_SI, const float3_64& cellSize_SI)
-                    {
-                        // TODO: respect bounding box
-
-                        {{#profile.type.uniform}}
-                            // note: right hand side is double
-                            return {{{profile.data.density_si}}} / SI::BASE_DENSITY_SI;
-                        {{/profile.type.uniform}}
-                    }
-                };
-
-                using init_{{{placed_species_initial.typename}}} = FreeFormulaImpl<init_{{{placed_species_initial.typename}}}_functor>;
-            {{/species_initmanager.operations.simple_density}}
-        } // namespace pypicongpu
-    } // namespace densityProfiles
+                    //! @todo respect bounding box, Brian Marre, 2023
+
+                    // one species only ever has exactly one profile, therefore only one of the blocks
+                    //      below will ever be present
+
+                    {{#profile.type.uniform}}
+                        return static_cast<float_X>({{{profile.data.density_si}}} / SI::BASE_DENSITY_SI);
+                    {{/profile.type.uniform}}
+
+                    {{#profile.type.foil}}
+                        float_64 const y = position_SI.y(); // m
+
+                        // begin & end plateau of foil
+                        constexpr float_64 y0 = {{{profile.data.y_value_front_foil_si}}}:
+                        constexpr float_64 y1 = y0 + {{{profile.data.thickness_foil_si}}};
+
+                        // prePlasma ramp
+                        {{#profile.data.pre_foil_plasmaRamp.type.exponential}}
+                            // exponential pre-expanded density ramp before plateau of foil
+                            constexpr float_64 prePlasmaLength = {{{profile.data.pre_foil_plasmaRamp.data.PlasmaLength}}};
+                            constexpr float_64 prePlasmaCutoff = {{{profile.data.pre_foil_plasmaRamp.data.PlasmaCutoff}}};
+
+                            // prefoil upramp
+                            if((y < y0) &&  y > y0 - prePlasmaCutoff)
+                                dens = static_cast<float_X>({{{profile.data.density_si}}} / SI::BASE_DENSITY_SI)
+                                    * math::exp((y - y0) / prePlasmaLength);
+                        {{/profile.data.pre_foil_plasmaRamp.type.exponential}}
+
+                        {{#profile.data.pre_foil_plasmaRamp.type.none}}
```

Comment:
Want to leave an explicit remark in the generated param-file that the no plasma ramp setting was used:

Helps to distinguish it from the failed pre- and post- ramp translation and following generation, which can lead to an empty string being substituted instead of generated block

Reply:
fine with me